src/key.rs

```diff
@@ -75,13 +73,20 @@ impl RedisKey {
         self.key_inner == null_key
     }
 
-    pub fn read(&self) -> Result<Option<String>, RedisError> {
-        let val = if self.is_null() {
-            None
+    pub fn read<'a>(&self) -> Result<Option<&'a [u8]>, RedisError> {
```

Comment:
```suggestion
    pub fn read(&self) -> Result<Option<&'_ [u8]>, RedisError> {
```
Not sure whats the `'a` is